src/transformer_deploy/convert.py

```diff
@@ -297,27 +317,82 @@ def get_pytorch_infer(model: PreTrainedModel, cuda: bool, task: str):
+                logger=trt_logger,
+                min_shape=tensor_shapes[0],
+                optimal_shape=tensor_shapes[1],
+                max_shape=tensor_shapes[2],
+                workspace_size=commands.workspace_size * 1024 * 1024,
+                fp16=not commands.quantization,
+                int8=commands.quantization,
+            )
+            save_engine(engine=encoder_engine, engine_file_path=tensorrt_encoder_path)
+            # check encoder engine has been correctly serialized
+            tensorrt_encoder: Callable[[Dict[str, torch.Tensor]], Dict[str, torch.Tensor]] = load_engine(
+                runtime=runtime, engine_file_path=tensorrt_encoder_path
+            )
+            del encoder_engine, runtime
+            decoder_onnx_path = os.path.join(commands.output, "t5-dec-if-node") + "/model.onnx"
+            tensorrt_decoder_path = os.path.join(commands.output, "t5-dec-if-node") + "/model.plan"
+            input_shapes = prepare_input_shapes_tensorrt_decoder(input_ids, model_pytorch.config.num_layers)
+            runtime: Runtime = trt.Runtime(trt_logger)
+            decoder_engine: ICudaEngine = build_engine(
+                runtime=runtime,
+                onnx_file_path=decoder_onnx_path,
+                logger=trt_logger,
+                input_shapes=input_shapes,
+                workspace_size=commands.workspace_size * 1024 * 1024,
+                fp16=not commands.quantization,
+                int8=commands.quantization,
+            )
+            save_engine(engine=decoder_engine, engine_file_path=tensorrt_decoder_path)
+            # check decoder engine has been correctly serialized
+            tensorrt_decoder: Callable[[Dict[str, torch.Tensor]], Dict[str, torch.Tensor]] = load_engine(
+                runtime=runtime, engine_file_path=tensorrt_decoder_path
+            )
+        else:
+            engine: ICudaEngine = build_engine(
+                runtime=runtime,
+                onnx_file_path=onnx_model_path,
+                logger=trt_logger,
+                min_shape=tensor_shapes[0],
+                optimal_shape=tensor_shapes[1],
+                max_shape=tensor_shapes[2],
+                workspace_size=commands.workspace_size * 1024 * 1024,
+                fp16=not commands.quantization,
+                int8=commands.quantization,
+            )
+            save_engine(engine=engine, engine_file_path=tensorrt_path)
+            # important to check the engine has been correctly serialized
+            tensorrt_model: Callable[[Dict[str, torch.Tensor]], Dict[str, torch.Tensor]] = load_engine(
+                runtime=runtime, engine_file_path=tensorrt_path
+            )
 
         if commands.task == "question-answering":
             tensorrt_inf: Callable[[Dict[str, torch.Tensor]], List[torch.Tensor]] = lambda x: list(
                 tensorrt_model(x).values()
             )
+        elif commands.generative_model == "t5":
```

Comment:
Instead of adding another condition here and creating t5_tensorrt_inference, maybe we could try to have a "tensorrt_model" defined in the previous step.

Reply:
I created the model before as you said.